### DISCUSSION TITLE
What do the special symbols in aaSeqCDR3 mean?

### DISCUSSION THREAD

**Discussion by li1311139481** [milaboratory/mixcr#1622] [Q&A] [answered]
I used mixcr to compare my TCR library, and in the results, I found many amino acid sequences representing CDR3 in the aaSeqCDR3 column, but I found many amino acid sequences containing _ and *, I don't understand what this means
For example:
```
aaSeqCDR3
SAASAYQGDRALIF   
CVLSWE_NNRILF    
CAWVRR_GRALIF    
CALAPRG_SGNKLIF  
CAVRDFHE_YNQWKLIF
CAAEPEGADRLTF    
CATDVGYAQGLTF    
CAVSDWPGNTGKLIF  
CAVTQ*IL_GLSGKLTF
CAFNSGTYQRF      
CALGRGS_VARSWF   
CALSGTTGGNNKLTF  
CALSEL*_HGKLIF   
CAVSFYTEGADRLTF  
CAMEVDHANKMIF    
CAMELGTGYQNFYF   
CAVSMDTGANTGKLMF 
CAVGEDYAHGLTF    
CAVSTPSNTNKVVF   
```

**Comment by mizraelson** [comment, marked as answer]
Hi,
`_` represents a frame shift
`*` is a stop codon

If the sequence has `_` or `*`, it is a nonfunctional clone that cant be translated normally.

**Reply by li1311139481** [reply]
Thanks for your reply
1. I can understand the identification of the stop codon, which simply means that when the translation goes here, the corresponding codon is one of the stop codons. However, there is some confusion about the frameshift represented by _. The amino acids after _ represent the frameshift mutated reads that would theoretically translate the frameshift amino acids. Or in the process of alignment, you would think like mRNA splinters, where the amino acid after _ is the corresponding amino acid that TRBD should translate correctly, where _ just means that frameshift occurred.
2. I found that despite matching the same TRBD and TRBJ regions, most aaSeqCDR3 is still very different. And there was no indication of a base mutation or frameshift mutation. Can you explain why that is?
3. The higher the HitsWithScore, the better the match between reads and genes？


<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///D:/SXruanjian/MobaXterm/data/slash/var/log/xwin/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///D:/SXruanjian/MobaXterm/data/slash/var/log/xwin/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


allDHitsWithScore | allJHitsWithScore | aaSeqCDR3
-- | -- | --
TRBD1*00(65) | TRBJ1-1*00(223.2) | CASSQGTGGADTEVFF
TRBD1*00(35) | TRBJ1-1*00(219.6) | CASSLFRGFTEVFF
TRBD2*00(40),TRBD1*00(35) | TRBJ1-1*00(218.2) | CASSQEGGGTEVFF
TRBD1*00(65) | TRBJ1-1*00(222.2) | CASSQWTGGADTEVFF
TRBD1*00(49) | TRBJ1-1*00(222) | CASSQGTWGADTEVFF
TRBD1*00(65) | TRBJ1-1*00(198.2) | CASRQGTG_R*HRSLF
TRBD1*00(49) | TRBJ1-1*00(222.9) | CASSQGTGCADTEVFF
TRBD1*00(25) | TRBJ1-1*00(214.4) | CASSLEAGNTEVFF
TRBD1*00(65) | TRBJ1-1*00(205.5) | CASSQGTGGADTEFFF
TRBD1*00(25) | TRBJ1-1*00(244.2) | CASSQDAGANTEVFF
TRBD1*00(25) | TRBJ1-1*00(239.3) | CASGVDTNTEVFF
TRBD1*00(65) | TRBJ1-1*00(199.2) | CASSQGTG_ALTQKSSF
TRBD1*00(25) | TRBJ1-1*00(218.5) | CASSQDA_GKHRSLF
TRBD1*00(49) | TRBJ1-1*00(222) | CASSQGTVGADTEVFF
TRBD1*00(65) | TRBJ1-1*00(224.3) | CASSQRTGGADTEVFF
TRBD1*00(65) | TRBJ1-1*00(223.7) | CVSSQGTGGADTEVFF
TRBD1*00(65) | TRBJ1-1*00(220.8) | FASSQGTGGADTEVFF
TRBD1*00(50) | TRBJ1-1*00(223.5) | CASSQGTGGADTEVFF
TRBD1*00(70) | TRBJ1-1*00(223.9) | CASSQGTGGADTEVFF
TRBD1*00(65) | TRBJ1-1*00(207.5) | CASSQGTGGADTEVFF
TRBD1*00(60) | TRBJ1-1*00(221.1) | CASSQVTGGADTEVFF
TRBD1*00(55) | TRBJ1-1*00(223) | CASSQGTGGTDTEVFF
TRBD1*00(65) | TRBJ1-1*00(223.3) | CANSQGTGGADTEVFF
TRBD1*00(65) | TRBJ1-1*00(203.8) | CASSQGTGGADT*VFF
TRBD1*00(25) | TRBJ1-1*00(223.7) | CASSQDAG_QTQKSSF
TRBD1*00(65) | TRBJ1-1*00(205) | CASSQGTGGADTEVLF
TRBD1*00(65) | TRBJ1-1*00(222.9) | RASSQGTGGADTEVFF
TRBD1*00(65) | TRBJ1-1*00(198.5) | CASSQGTG_R*HRSLL
TRBD1*00(65) | TRBJ1-1*00(187.7) | CASSQGTG_ALTQKSSL
TRBD1*00(65) | TRBJ1-1*00(198.3) | CASSQGTG_R*HKVFF
TRBD1*00(65) | TRBJ1-1*00(198.5) | CASSQGTG_R*HRSLF
TRBD1*00(65) | TRBJ1-1*00(187.2) | CASSQGTG_ALTQKSSF
TRBD1*00(35),TRBD2*00(30) | TRBJ1-1*00(219.5) | CASSPPGGGTEVFF
TRBD1*00(55) | TRBJ1-1*00(208.6) | CASSQGTGGADTEVFF



</body>

</html>

**Reply by mizraelson** [reply]
Hi,
1) A `CDR3` region must start and end with a complete codon; otherwise, the J and C gene sequences will be translated incorrectly. An _ indicates that a frameshift event has occurred because the number of nucleotides in the `CDR3` is not divisible by 3. Such mRNA can be translated but cannot be properly folded to form the receptor.

2) The CDR3 region is formed during the process of [VDJ recombination](https://en.wikipedia.org/wiki/V(D)J_recombination), where random nucleotides are added between V and D segments and between D and J segments. Also some nucleotides on the edges of the gene segments may be randomly trimmed. Thus, two clones may have the same genes involved in recombination, but their `CDR3` sequences will be different. 

3) The `score` is a characteristic of the alignment to the reference: the higher the proportion of the reference covered by the read with fewer mismatches, the higher the score will be.

**Reply by li1311139481** [reply]
Thanks a lot.
